src/codeManager.ts

```diff
@@ -238,8 +239,23 @@ export class CodeManager implements vscode.Disposable {
 
     private getExecutor(languageId: string, fileExtension: string): string {
         this._languageId = languageId === null ? this._document.languageId : languageId;
+
+        let executor = null;
+
+        // Check if file contains hash-bang
+        if (languageId == null) {

```

Comment:
languageId will always be not null, since https://github.com/turbulem/vscode-code-runner/blob/3e68fe166b6d98630df2f18ecbbc33963910daf2/src/codeManager.ts#L241

Reply:
But this line will change this._languageId and I'm checking parameter passed to this method. The idea is to try to call shebang executor when this method is called without params. It works fine for me, but maybe there is a better way to do this check?

BTW thank you for your time for reviews, very careful and helpful!

Reply:
Yes, you are right!